C7GameData/Tile.cs

```diff
@@ -55,12 +58,148 @@ public class Tile
 
 		public TileOverlays overlays = new TileOverlays();
 
+
+		// -1 = not initialized. 0 = false, 1 = true
```

Comment:
alternatively it could be a nullable boolean, i.e.

```
public bool? isLake = null;

public bool IsLake()
{
   if (isLake.HasValue) return isLake.Value;
   ...
}
```

Reply:
Also alternatively it can be a computed property of this class

```
public bool IsLake
{
   get
   {
      if (!IsWater()) return false;
      ...
   }
}
```